tutorials/tutorial5/tutorial-inverse-poisson.py

```diff
@@ -0,0 +1,117 @@
+import torch
+from torch.nn import Softplus
+from termcolor import colored
+import argparse
+from lightning.pytorch.callbacks import Callback
+from pina.problem import SpatialProblem, ParametricProblem, InverseProblem
+from pina.operators import laplacian
+from pina.model import FeedForward
+from pina.equation import Equation, FixedValue, ParametricEquation
+from pina import (Condition, CartesianDomain, PINN, LabelTensor,
+        Plotter, Location, Trainer)
+
+### Define ranges of variables
+x_min = -2
+x_max = 2
+y_min = -2
+y_max = 2
+### Import data (input and output that we consider as reference)
+data_output = torch.load('data/pinn_solution_0.5_0.5')
+data_input = torch.load('data/pts_0.5_0.5')
+
+### initialize the Poisson spatial problem
+class Poisson(SpatialProblem, InverseProblem):
+    '''
+    Problem definition for the Poisson equation.
+    '''
+    output_variables = ['u']
+    spatial_domain = CartesianDomain({'x': [x_min, x_max], 'y': [y_min, y_max]})
+    # initialize the parameters to be evaluated as torch Parameters
+    inferred_parameters = torch.nn.Parameter(torch.rand(2,
```

Comment:
this would be nice to not have it here. It is hard for a user who starts using PINA to understand the concept of `torch.nn.Parameter`. I was thinking to move it in [this function](https://github.com/annaivagnes/PINA/blob/15682571aab3d8b66aad3217e4a4a5a4b798f6af/pina/problem/abstract_problem.py#L99).

We can do something like:

```
    def _span_condition_points(self):
        """
        Simple function to get the condition points
        """
        for condition_name in self.conditions:
            condition = self.conditions[condition_name]
            if hasattr(condition, 'equation') and hasattr(condition, 'input_points'):
                samples = condition.input_points
            elif hasattr(condition, 'output_points') and hasattr(condition, 'input_points'):
                samples = (condition.input_points, condition.output_points)
            # skip if we need to sample
            elif hasattr(condition, 'equation') and isistance(condition.equation, ParametricEquation):
                  # here I initialize based on the domain the user gives me
            elif hasattr(condition, 'location'):
                self._have_sampled_points[condition_name] = False
                continue
            self.input_pts[condition_name] = samples
```